app/Helpers/database/leaderboard.php

```diff
@@ -75,9 +76,60 @@ function SubmitLeaderboardEntry(
         $retVal['BestScore'] = $newEntry;
     }
 
-    $retVal['TopEntries'] = GetLeaderboardEntriesDataJSON($lbID, $user, 10, 0, false);
-    $retVal['TopEntriesFriends'] = GetLeaderboardEntriesDataJSON($lbID, $user, 10, 0, true);
-    $retVal['RankInfo'] = GetLeaderboardRankingJSON($user, $lbID, (bool) $leaderboard->LowerIsBetter);
+    $retVal['RankInfo'] = [
+        'NumEntries' => $leaderboard->entries()->count(),
+    ];
+
+    if ($leaderboard->LowerIsBetter) {
+        $numBetterScores = $leaderboard->entries()->where('score', '<', $retVal['BestScore'])->count();
+        $retVal['RankInfo']['Rank'] = $numBetterScores + 1;
+
+        $entries = $leaderboard->entries()->orderBy('score');
+    } else {
+        // have to use <= for reverse sort so the number of users being subtracted includes
+        // all users with the same score (see issue #1201)
+        $numBetterScores = $leaderboard->entries()->where('score', '<=', $retVal['BestScore'])->count();
+        $retVal['RankInfo']['Rank'] = $retVal['RankInfo']['NumEntries'] - $numBetterScores + 1;
+
+        $entries = $leaderboard->entries()->orderByDesc('score');
+    }
+
+    $getEntries = function (Builder $query) {
+        $entries = $query
+            ->orderBy('updated_at')
+            ->with('user')
+            ->limit(10)
+            ->get()
+            ->map(fn ($entry) => [
+                'User' => $entry->user->User,
+                'Score' => $entry->score,
+                'DateSubmitted' => $entry->updated_at->unix(),
+            ])
+            ->toArray();
+
+        $index = 1;
+        $rank = 0;
+        $score = null;
+        foreach ($entries as &$entry) {
+            if ($entry['Score'] !== $score) {
+                $score = $entry['Score'];
+                $rank = $index;
+            }
+
+            $entry['Rank'] = $rank;
+            $index++;
+        }
```

Comment:
This `foreach` is the actual fix. Previously `$entry['Rank']` was just assigned to the index.